config/main.py

```diff
@@ -4413,6 +4549,107 @@ def delete(ctx):
     sflow_tbl['global'].pop('agent_id')
     config_db.set_entry('SFLOW', 'global', sflow_tbl['global'])
 
+#
+# set ipv6 link local mode on a given interface
+#
+def set_ipv6_link_local_only_on_interface(config_db, interface_dict, interface_type, interface_name, mode):
+
+    curr_mode = config_db.get_entry(interface_type, interface_name).get('ipv6_use_link_local_only')
+    if curr_mode is not None:
+        if curr_mode == mode:
+            return
+    else:
+        if mode == "disable":
+            return
+
+    if mode == "enable":
+        config_db.mod_entry(interface_type, interface_name, {"ipv6_use_link_local_only": mode})
+        return
+
+    # If we are disabling the ipv6 link local on an interface, and if no other interface
+    # attributes/ip addresses are configured on the interface, delete the interface from the interface table
+    exists = False
+    for key in interface_dict.keys():
+        if not isinstance(key, tuple):
+            if interface_name == key:
+                #Interface bound to non-default-vrf do not delete the entry
+                if 'vrf_name' in interface_dict[key]:
```

Comment:
Any issue in setting ""ipv6_use_link_local_only":disable" irrespective of entry present or not?

Reply:
It is setting the "ipv6_use_link_local_only":disable" when INTERFACE table has some attributes like vrf_name like below
"INTERFACE|Ethernet0": {
    "ipv6_use_link_local_only": "disable",
    "vrf_name": "Vrf1"
}

If not, it will delete the whole INTERFACE table instead of setting ""ipv6_use_link_local_only":disable".